docs/C API.md

```diff
@@ -0,0 +1,127 @@
+
+## P4 extern entries
+
+Context itself is unable to alter entries in extern, with exception for deleting all of them. For this purpose exist `entry`
+objects, which operates on a single entry or all entries. Possible operations (not available for all externs) for such
+entries are:
+- **add** or **insert**: add a new entry which did not exist before for given key/index.
+- **update**: modify value of an existing entry or of all entries if key/index is not provided within entry.
+- **del**: remove an existing entry or all entries if key/index is not provided within entry.
+- **get**: get value of a single entry for provided key/index within entry.
+- **get_next**: get value and key/index of a next entry.
+
+Sample usage of a table entry is:
+```c
+nikss_table_entry_t entry;
+nikss_table_entry_init(&entry);
+
+/* Create entry */
+
+nikss_table_entry_add(&table_ctx, &entry);
+nikss_table_entry_free(&entry);
+```
+
+Table below summarizes externs and type of entries.
+
+| Extern           | Entry type                                                                            | Notes                                                                     |
+|------------------|---------------------------------------------------------------------------------------|---------------------------------------------------------------------------|
+| `ActionProfile`  | `nikss_action_selector_member_context_t`                                              | The same type as for `ActionSelector`.                                    |
+| `ActionSelector` | `nikss_action_selector_member_context_t`,<br/>`nikss_action_selector_group_context_t` | Groups and members are different things.                                  |
+| Clone session    | `nikss_clone_session_entry_t`                                                         |
+| `Counter`        | `nikss_counter_entry_t`                                                               |
+| `Digest`         | `nikss_digest_t`                                                                      |
+| `DirectCounter`  | `nikss_direct_counter_entry_t`                                                        | 
+| `DirectMeter`    | `nikss_direct_meter_entry_t`                                                          | 
+| `Meter`          | `nikss_meter_entry_t`                                                                 |
+| Multicast group  | `nikss_mcast_grp_member_t`                                                            |
+| `Register`       | `nikss_register_entry_t`                                                              |
+| Table            | `nikss_table_entry_t`                                                                 |
+| `value_set`      | `nikss_table_entry_t`                                                                 | Due to https://github.com/NIKSS-vSwitch/nikss/issues/71 might be changed. |
```

Comment:
```suggestion
| `value_set`      | `nikss_table_entry_t`                                                                 | It might be changed (see https://github.com/NIKSS-vSwitch/nikss/issues/71). |
```

Reply:
I will remove this note, because #71 has been merged without significant changes.